internal/provider/custom_domain_verification_resource.go

```diff
@@ -0,0 +1,184 @@
+type customDomainVerificationResource struct {
+	client *propelauth.PropelAuthClient
+}
+
+// projectInfoResourceModel describes the resource data model.
+type customDomainVerificationResourceModel struct {
+	Environment types.String `tfsdk:"environment"`
+	Domain types.String `tfsdk:"domain"`
+	Timeouts timeouts.Value `tfsdk:"timeouts"`
+}
+
+func (r *customDomainVerificationResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_custom_domain_verification"
+}
+
+func (r *customDomainVerificationResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		Description: "Custom Domain Verification resource. This is for verifying a custom domain for Production or Staging.",
+		Attributes: map[string]schema.Attribute{
+			"environment": schema.StringAttribute{
+				Required: true,
+				Validators: []validator.String{
+					stringvalidator.OneOf("Staging", "Prod"),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Description: "The environment for which you are configuring the custom domain. Accepted values are `Staging`, `Prod`.",
+			},
+			"domain": schema.StringAttribute{
+				Required: true,
+				Description: "The domain to verify.",
+			},
+			"timeouts": timeouts.Attributes(ctx, timeouts.Opts{
+				Create: true,
+				Update: true,
+			}),
+		},
+	}
+}
+
+func (r *customDomainVerificationResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*propelauth.PropelAuthClient)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *propelauth.PropelAuthClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *customDomainVerificationResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan customDomainVerificationResourceModel
+
+	// Read Terraform plan data into the model
+	diags := req.Plan.Get(ctx, &plan)
+    resp.Diagnostics.Append(diags...)
+    if resp.Diagnostics.HasError() {
+        return
+    }
+
+	createTimeout, err := plan.Timeouts.Create(ctx, 15*time.Minute)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating a timeout", "Could not create a timeout for the custom domain verification.")
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, createTimeout)
+	defer cancel()
+
+	// Verify the custom domain
+	environment := plan.Environment.ValueString()
+	verificationErr := r.client.VerifyCustomDomainInfo(environment, false)
```

Comment:
Here we need to leverage the timeout created above. We should add some logic that parses out the specific error_code that indicates the TXT/CNAME record was not found (which likely means the DNS provider hasn't finished propagating it) and retries verification at intervals for the duration of the timeout.